### PR TITLE
Add an asychronous email queue for registration

### PR DESCRIPTION
**Pull request by ehsanmasdar**
Wraps email sending functionality in an asychronous queue so we don't overwhelm the server when sending large numbers of emails. If running registration with replicas, they all work on a shared email queue.

### CODE REVIEW

File: server/routes/api/user.ts [hunk 1]
```diff
@@ -244,17 +244,12 @@ function postApplicationBranchHandler(anonymous: boolean): (request: express.Req
...
 				if (!user.applied) {
-					await sendMailAsync({
-						from: config.email.from,
-						to: user.email,
-						subject: emailSubject || defaultEmailSubjects.apply,
-						html: emailHTML,
-						text: emailText
+					await agenda.now('send_templated_email', {
+						id: user.uuid,
+						subject: emailSubject || defaultEmailSubjects.preConfirm,
```
Comment (petschekr) [inline server/routes/api/user.ts]: Bad copy + paste: should be `defaultEmailSubjects.apply`

File: server/routes/api/user.ts [hunk 2]
```diff
@@ -244,17 +244,12 @@ function postApplicationBranchHandler(anonymous: boolean): (request: express.Req
...
-
 			if (questionBranch instanceof Branches.ApplicationBranch) {
 				if (!user.applied) {
-					await sendMailAsync({
-						from: config.email.from,
-						to: user.email,
-						subject: emailSubject || defaultEmailSubjects.apply,
-						html: emailHTML,
-						text: emailText
+					await agenda.now('send_templated_email', {
```
Comment (petschekr) [inline server/routes/api/user.ts]: Single quotes -> double quotes

File: server/routes/api/user.ts [hunk 3]
```diff
@@ -280,12 +275,10 @@ function postApplicationBranchHandler(anonymous: boolean): (request: express.Req
...
 				if (!user.confirmed) {
-					await sendMailAsync({
-						from: config.email.from,
-						to: user.email,
-						subject: emailSubject || defaultEmailSubjects.attend,
-						html: emailHTML,
-						text: emailText
+					await agenda.now('send_templated_email', {
+						id: user.uuid,
+						subject: emailSubject || defaultEmailSubjects.preConfirm,
```
Comment (petschekr) [inline server/routes/api/user.ts]: Should be `defaultEmailSubjects.attend`

File: server/routes/api/user.ts [hunk 4]
```diff
@@ -280,12 +275,10 @@ function postApplicationBranchHandler(anonymous: boolean): (request: express.Req
 
 			} else if (questionBranch instanceof Branches.ConfirmationBranch) {
 				if (!user.confirmed) {
-					await sendMailAsync({
-						from: config.email.from,
-						to: user.email,
-						subject: emailSubject || defaultEmailSubjects.attend,
-						html: emailHTML,
-						text: emailText
+					await agenda.now('send_templated_email', {
```
Comment (petschekr) [inline server/routes/api/user.ts]: Single quotes -> double quotes

File: server/routes/api/user.ts [hunk 5]
```diff
@@ -462,25 +454,18 @@ userRoutes.route("/send_acceptances").post(isAdmin, async (request, response): P
...
 			}
 
-			let html = await renderEmailHTML(emailMarkdown, user);
-			let text = await renderEmailText(emailMarkdown, user);
-
-			emails.push({
-				from: config.email.from,
-				to: user.email,
+			user.preConfirmEmailSent = true;
+			await agenda.now('send_templated_email', {
```
Comment (petschekr) [inline server/routes/api/user.ts]: Single quotes -> double quotes

**Comment by ehsanmasdar** [comment]
@petschekr I've tested locally that all 5 entrypoints send email correctly

**Pull request merged**